orchagent/watermarkorch.cpp

```diff
@@ -94,6 +85,52 @@ void WatermarkOrch::doTask(Consumer &consumer)
     }
 }
 
+void WatermarkOrch::handleWmConfigUpdate(const std::string &key, const std::vector<FieldValueTuple> &fvt)
+{
+    if (key == "TELEMETRY_INTERVAL")
+    {
+        for (std::pair<std::basic_string<char>, std::basic_string<char> > i: fvt)
+        {
+            if (i.first == "interval")
+            {
+                auto intervT = timespec { .tv_sec = to_uint<uint32_t>(i.second.c_str()) , .tv_nsec = 0 };
+                m_telemetryTimer->setInterval(intervT);
+
+            }
+            else
+            {
+                SWSS_LOG_WARN("Unsupported key: %s", i.first.c_str());
+            }
+        }
+    }
+}
+
+void WatermarkOrch::handleFcConfigUpdate(const std::string &key, const std::vector<FieldValueTuple> &fvt)
+{
+
+    if (key == "QUEUE_WATERMARK" || key == "PG_WATERMARK")
+    {
+        for (std::pair<std::basic_string<char>, std::basic_string<char> > i: fvt)
+        {
+            if (i.first == "FLEX_COUNTER_STATUS")
+            {
+                if (key == "QUEUE_WATERMARK")
+                {
+                    m_qWmEnabled = (i.second == "enable");
+                }
+                else if (key == "PG_WATERMARK")
+                {
+                    m_pgWmEnabled = (i.second == "enable");
+                }
+            }
+        }
+        if (isTimerEnabled())
+        {
+            m_telemetryTimer->start();
```

Comment:
If the telemetry timer has been started, a call to SelectableTimer::start() will reset the timer settings. This resets the current time to expiration, causing the interval length, where an enable cli issued, to be different from others. We should refrain from calling SelectableTimer::start() if there is no change to the timer interval value itself.

Reply:
Added logic to avoid uneven intervals.